### PR TITLE
update to yeoman-generator v0.17

### PR DESCRIPTION
**Pull request by code-vicar**
Hello,
I was interested in composing this generator with another one so I thought I'd have a go at updating this to version 0.17.

The functionality is unchanged except for a new option to allow a target destination to be specified.

I added some tests for the new option, and refactored the existing ones since the arguments weren't being applied for some reason.

I tried to follow the existing style, let me know what you think.

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -2,161 +2,159 @@
...
+
+		var destPath = this.destinationRoot();
+		if ( typeof this.options.destination === 'string' ) {
+			destPath = path.join(destPath, this.options.destination);
+		} else if ( this.options.destination === true ) {
+			destPath = path.join(destPath, configs.server);
+		}
+
+		// Only the `node` server configs require the `package.json` file
+		if ( configs.server !== 'node' ) {
```
Comment (sindresorhus) [inline app/index.js]:
``` js
if (configs.server !== 'node') {
```

Applies to every parens.

File: app/index.js [hunk 2]
```diff
@@ -2,161 +2,159 @@
...
-            fetch(props.choice);
-        });
-
-    }
-
-};
+var choices = [{
+	name: 'Apache',
+	value: {
+		'server': 'apache',
```
Comment (sindresorhus) [inline app/index.js]: Don't quote the key names.

File: app/index.js (hunk 1 above)
Comment (code-vicar) [inline app/index.js]: You would prefer no spaces inside the parens?  That's normally how I do it, but I thought I saw a contribution guideline somewhere that suggested these spaces.  I'll remove them now.

File: app/index.js [hunk 3]
```diff
@@ -2,161 +2,159 @@
...
+				'bower.json',
+				'doc',
+				'src',
+				'test'
+			]
+		};
+
+		if (this.server !== undefined) {
+			// There is no "pretty" built-in ability to break in `forEach, so
+			// use `some`: http://ecma-international.org/ecma-262/5.1/#sec-15.4.4.17
```
Comment (sindresorhus) [inline app/index.js]: comment not needed

Comment (arthurvr) [inline app/index.js]: The link to the spec isn't needed, but I like that comment being there. _At the end that's personal opinion._

Comment (sindresorhus) [inline app/index.js]: Code comments is not the place to teach JS.

Comment (code-vicar) [inline app/index.js]: Sure, I'll remove it.

**Comment by artuuu** [comment]
> . I'il review this shortly

The lame ass @arthurvr didn't review it, he just likes to comment to seem important and closes issues like he;s a big shot ... :-1:

**Comment by arthurvr** [comment]
@sindresorhus I think this is ready to land now?

**Comment by sindresorhus** [comment]
Thanks @code-vicar :)

**Comment by arthurvr** [comment]
Thanks @code-vicar!

**Pull request closed without merging**